pkg/arvo/lib/trie.hoon

```diff
@@ -0,0 +1,29 @@
+|%
```

Comment:
This doesn't appear to be used. Is it relevant to this PR?

Reply:
There were two times that I wanted a trie, so I wrote one.  One use case was an attempt at partial caching of the parsing, but the gains were only about 20% so I nixed it.  A trie would be the right data structure for filtering the identifiers in your subject according to the prefix you've typed, but I haven't used it because so far a list is fast enough.

As it's unused, I've removed it.  If we need a trie in the future, we can pull it out of git history.

Reply:
Makes sense. The entire autocomplete flow seems ripe for optimization and caching. But it's impressively fast already.